store/stat_store.go

```diff
@@ -0,0 +1,442 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"container/list"
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+	"time"
+)
+
+func NewStatStore(e uint64, r time.Duration, w uint8, sp []float64) *statStore {
```

Comment:
Without looking at the code, it's hard to tell what each argument does. And in generated docs, since the fields are not exported, it will be for the user to guess.

I suggest something like `NewStatStore(epsilon uint64, resolution time.Duration, duratoin uint8, percentiles []float64)`